### PR TITLE
chore(quality): tooling, types, robustness and a11y sweep

### PR DESCRIPTION
**Pull request by JacoboSanchez**
## Summary

Series of small, focused commits that raise the code-quality floor of the
repo without architectural rewrites. Each commit is independently
revertable. All gates pass: `ruff`, `mypy`, `pytest --cov=app
--cov-fail-under=70` (now at 82%), `npm run lint`, `npm run typecheck`,
`npm test`.

## Commits (in order)

1. **`feat(tooling): add ESLint + Prettier to the frontend`** — flat
   config with `typescript-eslint`, `react`, `react-hooks`, `jsx-a11y`,
   plus Prettier and `eslint-config-prettier`. Wired into CI (frontend
   job) and pre-commit (`local` hook on changed `.ts`/`.tsx`). Pre-existing
   dialog a11y findings and unused-import noise kept as warnings to avoid
   a flag day; later commits tighten them.
2. **`chore(repo): add .editorconfig, .gitattributes, CONTRIBUTING and templates`**
   — LF endings, indentation per language, binary classification,
   `linguist-generated` markers for lockfiles and the generated
   OpenAPI types. PR + bug + feature templates. `dependabot.yml` now
   tracks `npm` updates for `frontend/` (grouped by eslint / vitest /
   testing-library / react).
3. **`chore(quality): tighten Ruff and TypeScript strictness`** — Ruff
   rule set extended with `UP`, `C90` (cap 18), `RUF`, `SIM`. 199
   modernization auto-fixes (`Optional[X]` → `X | None`, `Dict[X, Y]`
   → `dict[X, Y]`, etc.). Per-file `C901` ignores for the two
   pre-existing god-functions (`match_report._compute_stats`, overlay
   router factory). **Real fix (RUF006):** `WSHub.broadcast_*_sync`
   now keeps a strong reference to its `loop.create_task` task so the
   loop doesn't GC the broadcast before it finishes. **Frontend:**
   `noUncheckedIndexedAccess`, `noUnusedLocals`, `noUnusedParameters`,
   `noImplicitReturns` enabled; new `DEFAULT_FONT_SCALE` export so
   `FONT_SCALES` fallback doesn't flow through indexed access; index
   sites in `useRecentEvents`, `useSwipeNavigation`, `i18n.tsx`,
   `theme.test.ts` hardened.
4. **`chore(quality): expand mypy coverage to ~17 modules`** —
   `[tool.mypy].files` grew from 6 to ~17 modules. Required minor
   cleanups (missing `@staticmethod` on `AppStorage` /
   `Customization` helpers, explicit type annotations on a few
   class-state dicts, fixed a wrong functional-Enum class name in
   `State.OIDStatus`). Per-module override silences `requests` /
   `websocket` import-untyped errors.
5. **`chore(quality): narrow silent except Exception blocks`** —
   `OverlayStateStore._read_state_sync`, `save_persisted_state[_async]`,
   `_iter_persisted_ids`, `_migrate_legacy_files_locked` now narrow
   to `OSError | json.JSONDecodeError`. `ObsBroadcastHub._send_to_clients`
   logs stale-client drops at debug level. `WSControlClient` close /
   heartbeat paths gain debug logs. **New tests** cover the
   corrupt-JSON and unreadable-file fallback paths.
6. **`chore(quality): drop unsafe casts and centralize schema types`**
   — Production `as unknown as Record<string, unknown>` cast in
   `useSettings::readAll` replaced with a typed `Object.assign`.
   `TeamState` centralized in `frontend/src/api/client.ts` so the
   four call sites stop reaching into `./schema` directly. Remaining
   `OverlayPreview.tsx` cast for the iframe `allowTransparency`
   attribute is now documented inline.
7. **`fix(a11y): make SetValueDialog and LinksDialog keyboard-accessible`**
   — New `frontend/src/components/Dialog.tsx` primitive with
   `role="dialog"`, `aria-modal="true"`, ESC-to-dismiss, and a real
   `<button>` backdrop (replaces the bare `<div onClick>` overlay).
   `SetValueDialog` and `LinksDialog` now compose the primitive.
   Same CSS classes / visual appearance — **no screenshot regen
   required.** Tests added for ESC and `aria-modal`.
8. **`test: backfill coverage for api.dependencies and bootstrap`** —
   24 tests for `app/api/dependencies` (`_strict_oid_access_enabled`,
   `check_oid_access` happy / 401 / 403 / lenient / strict / non-dict
   per-user, `get_current_username` edge cases) and 4 tests for
   `app/bootstrap.create_app` graceful behaviour when `frontend/dist`
   or `overlay_templates` are missing, plus `_split_csv_env`. Backend
   tests grew 853 → 883; coverage 82%.

## Test plan

- [x] `ruff check .`
- [x] `mypy` (39 source files type-checked, was 22)
- [x] `pytest tests/ --cov=app --cov-fail-under=70` — 883 passed, 82% coverage
- [x] `cd frontend && npm run lint` — 0 errors, 9 warnings (all
      pre-existing patterns tracked for follow-up cleanup)
- [x] `cd frontend && npm run typecheck` — clean
- [x] `cd frontend && npm test` — 327 passed (was 325)

## Notes for reviewers

- No operator-facing visual change. The dialog refactor preserves
  existing CSS classes and visual appearance, so screenshot
  regeneration is **not** needed.
- The Prettier `format:check` job was deliberately **not** wired into
  CI — running `prettier --write` would touch 63 files in pure
  reformatting noise. Scripts (`npm run format`, `npm run format:check`)
  are in place for local use; flipping the CI gate is a future cleanup.
- Each commit ships its own `CHANGELOG.md` entry under `[Unreleased]`.

https://claude.ai/code/session_01XJafveeCAWiyuSwB7xtBH1

---
_Generated by [Claude Code](https://claude.ai/code/session_01XJafveeCAWiyuSwB7xtBH1)_

### CODE REVIEW

File: frontend/src/components/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      }
+    };
+    document.addEventListener('keydown', onKey);
+    // Focus the dialog card so screen readers announce it and tab order
+    // starts inside the dialog rather than on the page below.
+    cardRef.current?.focus();
+    return () => {
+      document.removeEventListener('keydown', onKey);
+    };
+  }, [open, onClose]);
```
[the remark this answers is not in the document]
Comment (JacoboSanchez) [reply frontend/src/components/Dialog.tsx:85]: Good catch — fixed in 89a3cab.

I split the effect as you suggested (focus only on the `open` transition, keydown listener separately keyed on `open` + `onClose`) and went one step further on the focus-trap point: the keydown effect now also handles `Tab` / `Shift+Tab` and wraps focus within the card's focusable descendants, so keyboard users can't tab into the background page.

New regression test in `SetValueDialog.test.tsx`:

```ts
it('does not steal focus on parent re-render with unstable onClose', ...)
```

asserts that focusing the input and re-rendering with a fresh `onClose` arrow keeps focus on the input (was failing pre-fix).

---
_Generated by [Claude Code](https://claude.ai/code)_
